### ISSUE TITLE
Vue: ESLint of Vetur typescript errors isn't detected in the console when run `npm run serve`

### ISSUE THREAD

**Issue by aboelkassem** [typescript-eslint/typescript-eslint#6464]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

not included

### Repro Code

```typescript
private async created() {
    const test = blalblbl.find(x => x.id == 1); // blalblbl is undefined variable
}
```

### ESLint Config

```javascript
module.exports = {
  "parser": "vue-eslint-parser",
  "parserOptions": {
    "parser": "@typescript-eslint/parser"
  },
  rules: {},
  extends: [
      "plugin:vue/essential",
      "@vue/prettier",
      "@vue/typescript",
      "eslint:recommended",
      "plugin:vue/recommended"
    ],
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "ES5",
    "module": "esnext",
    "strict": true,
    "jsx": "preserve",
    "importHelpers": true,
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "sourceMap": true,
    "baseUrl": ".",
    "types": [
      "webpack-env"
    ],
    "paths": {
      "@/*": [
        "src/*"
      ]
    },
    "lib": [
      "esnext",
      "dom",
      "dom.iterable",
      "scripthost"
    ]
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "tests/**/*.ts",
    "tests/**/*.tsx"
  ],
  "exclude": [
    //"node_modules"
   ]
  }
}
```

### Expected Result

Show the semantic and syntax error while compiling in the console screen when run `npm run serve`

### Actual Result

It compiled successfully without any issues found in `.vue` files
![image](https://user-images.githubusercontent.com/42910789/218704643-ab5dff40-f9cd-46c8-946e-4b80d89f46e8.png)

### Additional Info

The error is detected with Vetur in the vscode
![image](https://user-images.githubusercontent.com/42910789/218704830-46c8865f-1908-465a-a5a3-09c48069fe09.png)

## additional packages info 
NodeJs version: v18.13.0
npm version: 8.19.3
"@typescript-eslint/eslint-plugin": "^5.1.0",
"@typescript-eslint/parser": "^5.1.0",
"@vue/cli-plugin-eslint": "^5.0.6",
"@vue/eslint-config-prettier": "^6.0.0",
"@vue/eslint-config-typescript": "^11.0.1",
"eslint": "^8.1.0",
"eslint-plugin-prettier": "^3.1.0",
"eslint-plugin-vue": "^9.4.0",
"vue-eslint-parser": "^9.1.0",

**Comment by bradzacher** [comment]
Separately solely based on your screenshots - your IDE only reports two eslint *warnings*, and the CLI says it has reported just warnings - so they look aligned and correct from our side - you're likely looking at problems in the Vue tooling (which isn't this project)
